server/src/main/java/org/elasticsearch/index/mapper/flattened/FlattenedFieldMapper.java

```diff
@@ -727,4 +730,40 @@ protected void parseCreateField(DocumentParserContext context) throws IOExceptio
     public FieldMapper.Builder getMergeBuilder() {
         return new Builder(simpleName()).init(this);
     }
+
+    @Override
+    public SourceLoader.SyntheticFieldLoader syntheticFieldLoader() {
+        return syntheticFieldLoader(simpleName());
+    }
+
+    protected SourceLoader.SyntheticFieldLoader syntheticFieldLoader(String simpleName) {
```

Comment:
Do we use this method? I know that `keyword` has one, but it has it because it calls it to make the `text` fields go. Otherwise I think I'd just combine this with the above method.